apps/sap-commerce-cloud-with-air/lambda/src/middlewares/getAppInstallationParameters.ts

```diff
@@ -0,0 +1,75 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express';
+import { getManagementToken } from '@contentful/node-apps-toolkit';
+import axios from 'axios';
+import { UnableToGetAppInstallationParameters } from '../errors/unableToGetAppInstallationParameters';
+import { AppInstallationParameters } from '../types/types';
+import { UnableToGetAppAccessToken } from '../errors/unableToGetAppAccessToken';
+
+export const getAppInstallationParametersMiddleware: RequestHandler = async (
+  req: Request,
+  _res: Response,
+  next: NextFunction
+) => {
+  try {
+    const installationParameters: AppInstallationParameters = {};
+
+    const appId = JSON.parse(JSON.stringify(req.header('x-contentful-app')));
+    const spaceId = JSON.parse(JSON.stringify(req.header('x-contentful-space-id')));
+    const environmentId = JSON.parse(JSON.stringify(req.header('x-contentful-environment-id')));
+    req.params.appId = appId;
+    req.params.spaceId = spaceId;
+    req.params.environmentId = environmentId;
+
+    if (!appId || !spaceId || !environmentId) {
+      throw new Error('Missing required headers!');
+    }
```

Comment:
just for my understanding of the SAP workflow, why are these required? I see why they were used in ecommerce project, this is to check credentials for GET requests prior to them saving the config (so then we can subsequently access via CMA?)?

Reply:
Excellent question. If you don't mind instead of writing this all out - I can just tuple with you and explain in person!